core_types/src/json.rs

```diff
@@ -145,38 +147,47 @@ pub trait JsonError {}
 
 impl JsonError for HolochainError {}
 
+fn result_to_json_string<T: Into<JsonString>, E: Into<JsonString>>(
+    result: Result<T, E>,
+) -> JsonString {
+    let is_ok = result.is_ok();
+    let inner_json: JsonString = match result {
+        Ok(inner) => inner.into(),
+        Err(inner) => inner.into(),
+    };
+    let inner_string = String::from(inner_json);
+    JsonString::from_json(&format!(
+        "{{\"{}\":{}}}",
+        if is_ok { "Ok" } else { "Err" },
+        inner_string
+    ))
```

Comment:
This looks correct but doesn't Result serialize to exactly this anyway? `serde_json::to_string(&result)`?

Reply:
i think that's the point right? to extend the types/traits so that they "magically" track default serde behaviour